### PR TITLE
feat(CPNT-2249):  Add Strict type rule

### PR DESCRIPTION
**Pull request by mohanrajpac**
- Adds declare strict type rule for classes
- Adds missing packages for the ruleset
- Removes deprecated rule https://github.com/humanmade/coding-standards/pull/246

### CODE REVIEW

File: NewsUK/ruleset.xml [hunk 1]
```diff
@@ -59,7 +59,8 @@
 		<exclude name="PSR2.Namespaces.UseDeclaration.MultipleDeclarations" />
 	</rule>
 	<rule ref="PSR2R.Namespaces.UnusedUseStatement" />
-	<rule ref="PSR2R.Namespaces.UseInAlphabeticalOrder" />
+
+	<rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes"/>
```
Comment (jonnynews) [inline NewsUK/ruleset.xml:63]:
```suggestion
	<rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes">
		<properties>
			<property name="declareOnFirstLine" value="true"/>
		</properties>
	</rule>
```

Comment (jonnynews) [reply NewsUK/ruleset.xml:63]: From https://github.com/slevomat/coding-standard/blob/d64610d04f204e4995b2c6918f1bd2ba959cfc3d/build/phpcs.xml#L443C1-L447C9

**Review by jonnynews — CHANGES_REQUESTED** [review]

File: composer.json [hunk 2]
```diff
@@ -16,6 +16,8 @@
     },
     "require": {
         "automattic/vipwpcs": "^3.0",
-        "dealerdirect/phpcodesniffer-composer-installer": "^1.0"
+        "dealerdirect/phpcodesniffer-composer-installer": "^1.0",
+        "phpcompatibility/phpcompatibility-wp": "^2.1",
+        "fig-r/psr2r-sniffer": "^2.0"
```
[the remark this answers is not in the document]
Comment (mohanrajpac) [reply composer.json:21]: It's a dependency for `psr2r-sniffer` so should be installed automatically. But it's good to be explicit.

File: NewsUK/ruleset.xml (hunk 1 above)
Comment (mohanrajpac) [reply NewsUK/ruleset.xml]: The explicit `declareOnFirstLine` conflicts with other rules. I would like to keep them as it is to ensure it checks for declare strict type exists in the file.

FOUND 3 ERRORS AFFECTING 3 LINES
---------------------------------------------------------------------------------------------------------------------------------------------------
 1 | ERROR | [ ] Missing file doc comment (Squiz.Commenting.FileComment.Missing)
 2 | ERROR | [x] There must be a single space between the PHP open tag and declare statement.
   |       |     (SlevomatCodingStandard.TypeHints.DeclareStrictTypes.IncorrectWhitespaceBeforeDeclare)
 4 | ERROR | [ ] The file-level docblock must follow the opening PHP tag in the file header (PSR12.Files.FileHeader.IncorrectOrder)

Comment (jonnynews) [reply NewsUK/ruleset.xml:63]: Good find!

**Pull request merged**